ios/Classes/SwiftAmplitudeFlutterPlugin.swift

```diff
@@ -1,252 +1,400 @@
+                return
+            }
+
+            do {
+                let event = try getEvent(args: args)
+                amplitude?.track(event: event)
+                amplitude?.logger?.debug(message: "Track event: \(String(describing: call.arguments))")
+
+                result("track called..")
+            } catch {
+                amplitude?.logger?.warn(message: "track called but failed.")
+            }
+
+        case "identify":
+            guard let args = call.arguments as? [String: Any] else {
+                print("\(call.method) called but call.arguments type casting failed.")
+                return
+            }
+
+            do {
+                let event = try getEvent(args: args)
+                amplitude?.track(event: event)
+                amplitude?.logger?.debug(message: "Track identify event: \(String(describing: call.arguments))")
+
+                result("identify called..")
+            } catch {
+                amplitude?.logger?.warn(message: "identify called but failed.")
+            }
+
+        case "groupIdentify":
+            guard let args = call.arguments as? [String: Any] else {
+                print("\(call.method) called but call.arguments type casting failed.")
+                return
+            }
+
+            do {
+                let event = try getEvent(args: args)
+                amplitude?.track(event: event)
+                amplitude?.logger?.debug(message: "Track group identify event: \(String(describing: call.arguments))")
+
+                result("groupIdentify called..")
+            } catch {
+                amplitude?.logger?.warn(message: "groupIdentify called but failed.")
+            }
+
+        case "setGroup":
+            guard let args = call.arguments as? [String: Any] else {
+                print("\(call.method) called but call.arguments type casting failed.")
+                return
+            }
+
+            do {
+                let event = try getEvent(args: args)
+                amplitude?.track(event: event)
+                amplitude?.logger?.debug(message: "Track set group event: \(String(describing: call.arguments))")
+
+                result("setGroup called..")
+            } catch {
+                amplitude?.logger?.warn(message: "setGroup called but failed.")
+            }
+
+        case "revenue":
+            guard let args = call.arguments as? [String: Any] else {
+                print("\(call.method) called but call.arguments type casting failed.")
+                return
+            }
+
+            do {
+                let event = try getEvent(args: args)
+                amplitude?.track(event: event)
+                amplitude?.logger?.debug(message: "Track revenue event: \(String(describing: call.arguments))")
+
+                result("revenue called..")
+            } catch {
+                amplitude?.logger?.warn(message: "revenue called but failed.")
+            }
```

Comment:
It looks like these event based sections are all same expect for logging. Could these maybe be consolidated into a single handler to make maintenance easier going forward?

```
case "track":
case "identify":
case "groupIdentify":
case "setGroup":
case "revenue":
   ...handler code
```

Reply:
Consolidate both Swift and Kotlin plugins at https://github.com/amplitude/Amplitude-Flutter/pull/176/commits/d0e5f078b78db6be343696b4de591d0cec2826d8

Reply:
Awesome thanks!